docs: improve stdout-log-generator README with full argument documentation (#3763)
kyma-project/telemetry-manager: dependencies/stdout-log-generator/README.md

# Stdout Log Generator

> A small tool that continuously writes log lines to stdout. Used in the e2e tests of telemetry-manager to produce configurable log traffic inside Kubernetes workloads.

Reviewer: ```suggestion
`stdout-log-generator` is a tool that continuously writes log lines to stdout. Telemetry Manager e2e tests use it to generate configurable log traffic inside Kubernetes workloads.
```

> The tool exposes a Prometheus metrics endpoint on port `2112` (`/metrics`) with two metrics:

Reviewer: ```suggestion
The tool exposes a Prometheus metrics endpoint on port `2112` (`/metrics`) with the following metrics:
```

never mention the exact number when introducing a list, easier for maintenancce

- `logs_generated_total` — total log lines written since start
- `logs_generated_rate` — actual throughput in logs/second since start

## Usage

```sh
stdout-log-generator [flags]
```

## Flags

| Flag | Short | Default | Description |
|------|-------|---------|-------------|
| `--format` | | `json` | Output format: `json` or `plaintext` |
| `--bytes` | `-b` | `2048` | Size of each log line in bytes |
| `--rate` | `-r` | `1` | Target log lines per second. `0` means unlimited |

> | `--fields` | `-f` | | Additional key=value fields to include in every JSON log record (can be repeated or comma-separated). Ignored when `--format plaintext` is used |

Reviewer: less ambiguity

```suggestion
| `--fields` | `-f` | | Additional key=value fields to include in every JSON log record. You can specify it multiple times or provide a comma-separated list. Ignored when `--format plaintext` is used |
```

> | `--text` | `-t` | | Fixed text to write for every plaintext log line. Ignored if `--bytes` is set. Only relevant when `--format plaintext` is used |

Reviewer: ```suggestion
| `--text` | `-t` | | Fixed text to write for every plaintext log line. Applies only if you use `--format plaintext`. Ignored if you set `--bytes`.  |
```

## Output formats

### JSON (default)

> Each line is a JSON object with a `padding` field filled with random characters to reach the requested byte size, plus any custom fields added via `--fields`:

Reviewer: ```suggestion
Each line is a JSON object with a `padding` field filled with random characters to reach the requested byte size, plus any custom fields added with `--fields`:
```

```json
{"padding":"aBcDeFgH...","environment":"prod","app":"myservice"}
```

The `padding` field is sized so that the total serialized JSON line matches `--bytes` exactly (including the trailing newline added by the JSON encoder).

If the combined size of the custom fields already exceeds `--bytes`, the tool exits with an error.

### Plaintext

> Each line is either the fixed string given by `--text`, or a random string of `--bytes` length when `--text` is not set.

Reviewer: ```suggestion
The content of each line depends on whether you set the `--bytes` flag:
- With `--bytes`, each line is a random string of the specified length.
- Without `--bytes` but with `--text`, each line is the fixed string you provide.
```

> 

Reviewer: what if neither flag (`--bytes` or `--text`) is set? Random string with default byte size of `2048`.

## Rate limiting

> The tool uses a token-bucket rate limiter (`golang.org/x/time/rate`). On Linux, the minimum achievable sleep granularity is approximately 2 ms. To compensate, the burst size is set to `2 × (rate / 1000)` so that the generator can emit multiple logs back-to-back before sleeping, keeping the long-run average close to the requested rate even at high throughputs.

Reviewer: Splitting the sentence for clarity

```suggestion
The tool uses a token-bucket rate limiter (`golang.org/x/time/rate`). On Linux, the minimum achievable sleep granularity is approximately 2 ms. To compensate for this, the burst size is set to `2 × (rate / 1000)`. This setting lets the generator emit multiple logs back-to-back before sleeping, keeping the long-run average close to the requested rate even at high throughput.
```

> Setting `--rate 0` disables throttling entirely (write as fast as possible).

Reviewer: purpose first, then instruction

```suggestion
To disable throttling and write logs as fast as possible, set `--rate 0`.
```
